### PR TITLE
Improve code coverage

### PR DESCRIPTION
**Pull request by chonguschonguschongus**
Add more test cases to improve code coverage as well as a DoctorUtil class for testing AddDoctorCommand and its parser.

Moving forwards, I will try to write more test cases for AddDoctorCommand and its parser.

### CODE REVIEW

File: src/test/java/seedu/address/logic/parser/KeywordParserTest.java [hunk 1]
```diff
@@ -15,24 +14,73 @@
...
 public class KeywordParserTest {
+    @Test
+    public void parseInputWithName() {
+        String[] input = {"John", "Doe"};
+        Predicate<Person> predicate = KeywordParser.parseInput(input);
+        assertTrue(predicate instanceof NameContainsKeywordsPredicate);
+    }
+
+    @Test
+    public void parseInputWithIC() {
```
Comment (tanveersingh10) [inline src/test/java/seedu/address/logic/parser/KeywordParserTest.java:25]: I think for these tests can try follow the convention where the name of the method should have the expected outcome and stuff like that.

File: src/test/java/seedu/address/model/person/GenderPredicateTest.java [hunk 2]
```diff
@@ -22,6 +23,41 @@ public void testGenderPredicate() {
...
+
+        GenderPredicate firstPredicate = new GenderPredicate(firstPredicateKeyword);
+        GenderPredicate secondPredicate = new GenderPredicate(secondPredicateKeyword);
+
+        // same object -> returns true
+        assertTrue(firstPredicate.equals(firstPredicate));
+
+        // same values -> returns true
+        GenderPredicate firstPredicateCopy = new GenderPredicate(firstPredicateKeyword);
+        assertTrue(firstPredicate.equals(firstPredicateCopy));
```
Comment (tanveersingh10) [inline src/test/java/seedu/address/model/person/GenderPredicateTest.java:42]: Good job adding comments to make code more readable

**Review by tanveersingh10 — APPROVED** [review]
lgtm! Made a small nit about the naming of test methods. Thanks for improving code coverage

**Review by cmHuang777 — APPROVED** [review]
Overall looks good, may be you can fix the issues mentioned, before merging into the master.

File: src/main/java/seedu/address/logic/parser/AddressBookParser.java [hunk 3]
```diff
@@ -41,7 +42,7 @@ public class AddressBookParser {
      * @return the command based on the user input
      * @throws ParseException if the user input does not conform the expected format
      */
-    public Command parseCommand(String userInput) throws ParseException {
+    public Command parseCommand(String userInput) throws ParseException, CommandException {
```
[the remark this answers is not in the document]
Comment (chonguschonguschongus) [reply src/main/java/seedu/address/logic/parser/AddressBookParser.java:45]: Good catch, thanks.

File: src/test/java/seedu/address/logic/parser/KeywordParserTest.java (hunk 1 above)
Comment (chonguschonguschongus) [reply src/test/java/seedu/address/logic/parser/KeywordParserTest.java:25]: Yes will make the edits

**Pull request merged**